packages/gestalt/src/Toast.js

```diff
@@ -112,69 +112,86 @@ export default function Toast({
-              </Text>
+                {thumbnail}
+              </Mask>
             </Flex.Item>
+          ) : null}
+          {isErrorVariant ? (
+            <Flex.Item flex="none">
+              <Icon
+                color="inverse"
+                icon="workflow-status-problem"
+                accessibilityLabel="problem"
+                size={32}
+              />
+            </Flex.Item>
+          ) : null}
+          <Flex.Item flex="grow">
+            <Text
+              weight={isErrorVariant ? 'bold' : undefined}
+              align={!thumbnail && !primaryAction ? 'center' : 'start'}
+              color={textColor}
+            >
+              {textElement}
+            </Text>
+          </Flex.Item>
 
-            {primaryAction || _dangerouslySetPrimaryAction ? (
-              // Allow button text to wrap on mobile
-              <Flex.Item flex={isMobileWidth ? 'shrink' : 'none'}>
-                {isValidElement(_dangerouslySetPrimaryAction) ? _dangerouslySetPrimaryAction : null}
-                {!_dangerouslySetPrimaryAction &&
-                primaryAction?.accessibilityLabel &&
-                primaryAction?.label ? (
-                  <ToastPrimaryAction
-                    accessibilityLabel={primaryAction.accessibilityLabel}
-                    href={primaryAction.href}
-                    rel={primaryAction?.rel}
-                    size="sm"
-                    target={primaryAction?.target}
-                    label={primaryAction.label}
-                    onClick={primaryAction.onClick}
-                  />
-                ) : null}
-              </Flex.Item>
-            ) : null}
-          </Flex>
-        </Box>
+          {primaryAction || _dangerouslySetPrimaryAction ? (
+            // Allow button text to wrap on mobile
+            <Flex.Item flex={isMobileWidth ? 'shrink' : 'none'}>
+              {isValidElement(_dangerouslySetPrimaryAction) ? _dangerouslySetPrimaryAction : null}
+              {!_dangerouslySetPrimaryAction &&
+              primaryAction?.accessibilityLabel &&
+              primaryAction?.label ? (
+                <ToastPrimaryAction
+                  accessibilityLabel={primaryAction.accessibilityLabel}
+                  href={primaryAction.href}
+                  rel={primaryAction?.rel}
+                  size="sm"
+                  target={primaryAction?.target}
+                  label={primaryAction.label}
+                  onClick={primaryAction.onClick}
+                />
+              ) : null}
+            </Flex.Item>
+          ) : null}
+        </Flex>
       </Box>
     );
+    return (
+      <Fragment>
+        <Box
+          // Ensure that maxWidth isn't greater than viewport width (for small screens)
+          maxWidth={`min(${EXP_TOAST_MAX_WIDTH}px, 100vw)`}
+          role="status"
+          rounding={4}
+          smDisplay="none"
+          width="100%" // only full width below the smallest breakpoint
```

Comment:
Can you elaborate on what's being solved here, maybe with screenshots? I'm not sure I follow why this update is needed